src/main/java/org/xbill/DNS/SimpleResolver.java

```diff
@@ -444,4 +444,80 @@ private Message sendAXFR(Message query) throws IOException {
   public String toString() {
     return "SimpleResolver [" + address + "]";
   }
+
+  private class Verify1Response {
+    private Message query;
+    private int qid;
+    private Throwable exc;
+
+    public Verify1Response(Message query, int qid) {
+      this.query = query;
+      this.qid = qid;
+    }
+
+    public Throwable getExc() {
+      return exc;
+    }
+
+    public Message parse(byte[] in) {
+      exc = null;
+
+      // Check that the response is long enough.
+      if (in.length < Header.LENGTH) {
+        exc = new WireParseException("invalid DNS header - too short");
```

Comment:
I didn't want to throw them unless there were no successful responses to a mDNS query. mDNS can get garbage packets back which should be silently ignored.And, because there could be multiple responses, I only wanted to throw the exception if no valid responses were returned (i.e., hacker returned garbage, then valid Avahi-daemon returned a legitimate answer [in that order], then the query shouldn't fail).

Reply:
Yes, I get that, but instead of returning null and then checking the exception instance from a getter, you can throw, catch and still continue in the mDNS case.